replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/data/ReplicationSiteImpl.java

```diff
@@ -89,21 +121,60 @@ public boolean isRemoteManaged() {
     Map<String, Object> result = super.toMap();
     result.put(NAME_KEY, getName());
     result.put(URL_KEY, getUrl());
+    final String vurl = getVerifiedUrl();
+
+    if (vurl != null) {
+      result.put(VERIFIED_URL_KEY, vurl);
+    }
     result.put(IS_REMOTE_MANAGED_KEY, isRemoteManaged());
     return result;
   }
 
   @Override
   public void fromMap(Map<String, Object> properties) {
     super.fromMap(properties);
+    final int version = super.getVersion();
+
+    if (version == ReplicationSiteImpl.CURRENT_VERSION) {
+      fromCurrentMap(properties);
+    } else {
+      fromIncompatibleMap(version, properties);
+    }
+  }
+
+  private void fromCurrentMap(Map<String, Object> properties) {
     setName((String) properties.get(NAME_KEY));
     setUrl((String) properties.get(URL_KEY));
+    setVerifiedUrl((String) properties.get(VERIFIED_URL_KEY)); // do this after setUrl()
+    setRemoteManaged(Boolean.parseBoolean((String) properties.get(IS_REMOTE_MANAGED_KEY)));
+  }
 
-    if (super.getVersion() == 1) {
-      setRemoteManaged(false);
-      super.setVersion(CURRENT_VERSION);
-    } else {
-      setRemoteManaged(Boolean.parseBoolean((String) properties.get(IS_REMOTE_MANAGED_KEY)));
+  private void fromIncompatibleMap(int version, Map<String, Object> properties) {
+    switch (version) {
+      case 1:
+        fromVersionOneMap(properties);
+        break;
+      case 2:
+        fromVersionTwoMap(properties);
+        break;
+      default:
```

Comment:
❓ Don't we have Sites without versions?

Reply:
Nvm forgot about OldSite. Just tested upgrading 0.2.0 to 0.2.3-SNAPSHOT and seemed to work fine.